src/dialog/an-dialog.factory.js

```diff
@@ -151,6 +160,53 @@
                         });
                     };
 
+                    var getFocusableElements = function() {
+                        var focusableEls = element.find('a[href], area[href], input:not([disabled]), select:not([disabled]), textarea:not([disabled]), button:not([disabled]), [tabindex="0"]');
+
+                        focusableElems = Array.prototype.slice.call(focusableEls);
+                    };
+
+                    var setInitialFocusableElements = function () {
+                        firstFocusableElem = focusableElems[0];
+                        lastFocusableElem = focusableElems[focusableElems.length - 1];
+                    };
+
+                    var onDialogKeyDownEvent = function(e) {
+                        var KEY_TAB = 9;
+
+                        switch(e.keyCode) {
+                            case KEY_TAB:
+                                if (firstFocusableElem.length === 1) {
+                                    e.preventDefault();
+                                    break;
+                                }
+
+                                if (e.shiftKey) {
+                                    onBackwardTab(e);
+                                } else {
+                                    onForwardTab(e);
+                                }
+
+                                break;
+                            default:
+                                break;
+                        }
+                    };
+
+                    var onBackwardTab = function(e) {
+                        if (document.activeElement === firstFocusableElem) {
+                            e.preventDefault();
+                            lastFocusableElem.focus();
+                        }
```

Comment:
Shouldn't the previous element get focused in the else case?

Reply:
Basically, the initial navigation through "tab" button should work there by default, because 
1) we don't have e.preventDefault() for "tab" button in default case
2) onBackwardTab and onForwardTab I cover only edge cases to loop "tab" action